### PR TITLE
Add authentication and authorization handy classes

### PR DESCRIPTION
**Pull request by batopa**
This PR add some classes that help with authentication and authorization.

Since these features are optional the `cakephp/authentication` and `cakephp/authorization` dependencies are only suggested, so you have to install them if needed.

### CODE REVIEW

File: src/Policy/RequestPolicy.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+     */
+    protected function applyRolesPolicy(IdentityInterface $identity, array $roles): Result
+    {
+        $identityRoles = (array)Hash::get((array)$identity->getOriginalData(), 'roles');
+
+        if (!empty(array_intersect($identityRoles, $roles))) {
+            return new Result(true);
+        }
+
+        return new Result(false, 'request forbidden for identity\'s roles');
```
Comment (didoda) [inline src/Policy/RequestPolicy.php:168]: minor: I'd invert (as in `missingRuleResult`) the logic:
```
if (empty(array_intersect($identityRoles, $roles))) {
    return new Result(false, 'request forbidden for identity\'s roles');
}

return new Result(true);
```

File: tests/TestCase/Policy/RequestPolicyTest.php [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    /**
+     * Test that rules work as expected.
+     *
+     * @param mixed $expected The expected result
+     * @param \Cake\Http\ServerRequest $request The request
+     * @param array $policyConfig Policy configuration
+     * @param array $identityData Identity data
+     * @return void
+     * @dataProvider canAccessProvider()
+     */
```
Comment (didoda) [inline tests/TestCase/Policy/RequestPolicyTest.php:302]: missing phpdoc `@covers ::canAccess()`

**Review by didoda — CHANGES_REQUESTED** [review]
minor changes requested

File: tests/TestCase/Policy/RequestPolicyTest.php (hunk 2 above)
Comment (batopa) [reply tests/TestCase/Policy/RequestPolicyTest.php:302]: SInce the test cover all class methods I'm going to add `@covers` to class doc block

**Review by didoda — APPROVED** [review]

**Review by didoda — APPROVED** [review]

**Pull request merged**